docs/enhancements/ai-on-zero-worker-hypershift.md

```diff
@@ -0,0 +1,118 @@
+
+<b>AgentServiceConfig CRD shall be converted to <i>Namespaced</i> scope (Rejected)</b>:
+Reusing AgentServiceConfig while converting its scope was rejected because the operator can not update existing CRs (scope is an immutable field)<br>
+<b>Deploy multiple AgentServiceConfig CRs on a cluster scope.</b>:
+The target namespace will be part of the Spec. When the Spec does not contain the target namespace and other hypershift-related fields, then the deployment is assumed to behave as usual.
+This option was rejected because it requires too much behavioral branching in the current implementation and requires taking care of edge cases such as detecting when the user deployed multiple CRDs in a non-hypershift environment and rejecting the such option. Also, having the namespace field in the Spec is a hack.
+<b>Selected Solution: New CRD HypershiftServiceConfig on a namespace scope</b>
+- Separation of intent
+- Natural namespacing
+
+### Assisted-service controllers working with hostedCluster (L1) control plane
+From the user point of view, they define the cluster installation CRs on one of the namespaces in the hypershift cluster, and the assisted service picks them up and spins the cluster. To achieve that, we need to:
+1. Have all cluster deployment CRDs defined on L1 
+2. Assisted service's controllers should work with L1 api-server (the hosted cluster) exclusively, even though the pods on which the controllers are running are hosted on L0 (the management cluster).
+
+#### CRD deployment
+This task, which is currently done by the OLM on the management cluster, should now carried by the operator.
+There are also 2 options for finding which CRDs we need to deploy:
+1. Build mechanism takes the CRDs from the deploy yaml and stores them on the filesystem of the operator
+2. The operator reads the current CRDs on the management system. Once the CRDs are known to the operator, it will use a runtime client pointed to the L1 api-server to register the CRDs
+
+#### Client Configuration
+In the assisted service, we define a KUBECONFIG env variable to the service deployment that points to a mounted secret containing L1's kubeconfig. This secret is defined in a well-defined place in the management cluster and is also used by ACM. In this way, the deployed service to namespace <b>k</b> communicates with its corresponding hypershift hosted cluster out of the box with no code changes required.
+
+
+Open Issue: How and where is that secret provisioned and become known to us (AI/ACM integration issue)
+
+### HypershiftServerConfig controller (Infrastructure operator)
+#### Client configuration
+EXTERNAL_KUBECONFIG  env variable to the service deployment points to a mounted secret containing L1's kubeconfig. This secret is defined in a well-defined place in the management cluster and is also used by ACM.
+
+The controller shall register itself to watch and manage resources with both L0 and L1 api-servers and instantiate 2 clients, one for each control plane.
```

Comment:
I thought we couldn't actually watch resources from our operator in L1.

Reply:
We must configure the CRDs, endpoint, services and RBAC there.